ui/src/main.mjs

```diff
@@ -72,7 +66,7 @@ const vuetify = createVuetify({
 // if our scoket disconnects, we should inform the user when it reconnects
 
 // GET our SocketIO Config from Node-RED & any other bits plugins have added to the _setup endpoint
-fetch('_setup')
+fetch('_setup', { mode: 'no-cors' })
```

Comment:
I'm not convinced by the no-cors mode because it has multiple security implications. I was able to bypass the blank screen without cors disabled, albeit I only tested on chrome w linux.

The proper solution would be to pass the specific access control origin headers. I'm not sure if we have access to that information when this call is being made. @joepavitt could weigh in on this

Reply:
Yeah, `no-cors` _shouldn't_ be necessary here because the `_setup` call is always called to the same host?

Reply:
After a chat with @joepavitt , it is my understanding that this _setup call will always be same origin as the NR instance serves the dashboard via an express server.

Cors might become an issue after adding the ability to host the dashboard on a separate domain than the NR instance, but we can mark that as a problem for future selves because it will bring up another host related of problems

Reply:
Does the same hold true if your auth service is on same host but different port and/or subdomain?

Quote from #817: 

> I use Authentik, a self-hosted auth provider with Google OAUTH as a social login option in front of NodeRed. In Edge on Windows I tried to replicate this issue by removing session cookies from Dashboard in an attempt to emulate session expiration. It did indeed act in the manner you described: blank screen on Dashboard (reloading dashboard didn't work), but navigating to NodeRed editor redirected to OAUTH login as it should. After logging in, Dashboard worked again.
> 
> Dev console in Edge showed the following error when Dashboard blank screen appeared: `Access to fetch at 'https://example_authentik_oauth_server.com' (redirected from 'https://example_nodered_instance.com/dashboard/_setup') from origin 'https://example_nodered_instance.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled. `
>

Reply:
This may be an isolated scenario with my deployment, we could remove `no-cors` and see if issue persists.

Reply:
I have removed the no-cors option from that line on the version I am testing to see if it is still ok. I will have to wait for my google auth to expire before I will know whether it still redirects to google ok.

Reply:
Thank you!

I have a feeling that the addition of the [Workbox](https://developer.chrome.com/docs/workbox/) lib and it's caching strategies is doing most of the heavy lifting when it comes to resolving auth cache problems but didn't get a chance to properly test it because the hole setup is quite extensive and in the small setup I was able to test it in seemed to work just fine.

Looking forward to your feedback!

Reply:
It appears that the CORS error no longer occurs when I conduct the same test without using the `no-cors` option. Although I’m not entirely sure what has changed, the test now works correctly. 

If @colinl doesn’t encounter any issues with his test, I believe we can safely proceed to remove the `no-cors` option without any adverse consequences.

Reply:
Glad to hear! 

> If @colinl doesn’t encounter any issues with his test, I believe we can safely proceed to remove the no-cors option without any adverse consequences.
Definately!

Reply:
Also tried without the `no-cors` line, and using the same setup as last week I am able to login in Windows (Edge) and Android (Chrome + PWA)

Reply:
> Also tried without the `no-cors` line, and using the same setup as last week I am able to login in Windows (Edge) and Android (Chrome + PWA)

That's very good news!

Reply:
On one of my Android devices and on Edge on the laptop it redirected to the google auth page and logged in ok.  The second Android device had the dashboard in the background. When I pulled it forward it showed the Loading screen with the array of flashing dots and stuck there.  I haven't seen that with the cors option included, but maybe it would happen occasionally with it, I don't know.  Pulling down to force a reload then redirected ok.  So at the moment I am not sure what to make of that.

Reply:
At any rate, in the interest of caution, I'd suggest removing the `no-cors` option and expand our testing to a larger audience to determine if any problems continue to occur.

Reply:
Agreed!